src/UI/Implementation/Component/Trigger/TriggerAction.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+
+namespace ILIAS\UI\Implementation\Component\Trigger;
+
+use Doctrine\Instantiator\Exception\InvalidArgumentException;
+
+/**
+ * @author Stefan Wanzenried <sw@studer-raimann.ch>
+ */
+class TriggerAction implements \ILIAS\UI\Component\Trigger\TriggerAction
+{
+
+    /**
+     * @var \ILIAS\UI\Component\Component
+     */
+    protected $component;
+
+    /**
+     * @var string
+     */
+    protected $event = 'click';
+
+    /**
+     * @var \Closure
+     */
+    protected $js_binding;
+
+    /**
+     * @var array
+     */
+    protected static $events = array('click', 'hover', 'change', 'dblclick');
```

Comment:
I would like this to be abstracted somewhat, e.g. by using constants that are accessible through the interface TriggerAction. E.g. TriggerAction::CLICK